modules/guides/pages/bulk-operations.adoc

```diff
@@ -0,0 +1,420 @@
+
+----
+
+{github}
+--
+
+Java::
++ 
+--
+. Add new data to update some existing JSON documents.
+
+. Using the `reactor.core.publisher.Flux` reactive library, call the `fromIterable()` method to perform multiple `upsert` or `replace` operations.
+
+{kv-bulk-update-example}
+
+[source,java]
+----
+include::java-sdk:hello-world:example$KvBulkHelloWorld.java[tag=kv-users,indent=0]
+
+include::java-sdk:hello-world:example$KvBulkHelloWorld.java[tag=kv-bulk-upsert,indent=0]
+
+----
+
+{github}
+--
+
+Node.js::
++ 
+--
+. Add new data to update some existing JSON documents.
+
+. Perform an `upsert` operation on each document and wait for all the promises to complete before accessing the results.
+
+{kv-bulk-update-example}
+
+[source,nodejs]
+----
+include::nodejs-sdk:hello-world:example$kv-bulk-hello-world.js[tags=kv-users;kv-bulk-upsert,indent=0]
+----
+
+{github}
+--
+
+Python::
++ 
+--
+. Add new data to update some existing JSON documents.
+
+. Pass the new document data to `upsert_multi()`. This will update all the documents in the database.
+
+. A dictionary of `MutationResult` objects is returned.
+
+CAUTION: `CBCollection.upsert_multi` is a xref:python-sdk:project-docs:compatibility.adoc#interface-stability[volatile] API call that is still in flux and may likely be changed.
+
+{kv-bulk-update-example}
+
+[source,python]
+----
+include::python-sdk:hello-world:example$kv_bulk_hello_world.py[tags=kv-users;kv-bulk-upsert,indent=0]
+----
+
+{github}
+--
+====
+
+== Deleting Multiple Documents
+
+To delete multiple documents in Couchbase perform a bulk `remove` operation.
+
+[{tabs}]
+====
+pass:[.NET]::
++ 
+--
+. Initialize a list of tasks.
+
+. Perform a `remove` operation on each document and store the results in the tasks list.
+
+. Wait for all the tasks to complete.
+
+{kv-bulk-delete-example}
+
+[source,csharp]
+----
+include::dotnet-sdk:hello-world:example$KvBulkHelloWorld.csx[tag=kv-users,indent=0]
+
+include::dotnet-sdk:hello-world:example$KvBulkHelloWorld.csx[tag=kv-bulk-remove,indent=0]
+
+----
+
+{github}
+--
+
+Java::
++ 
+--
+Using the `reactor.core.publisher.Flux` reactive library, call the `fromIterable()` method to perform multiple `remove` operations.
+
+{kv-bulk-delete-example}
+
+[source,java]
+----
+include::java-sdk:hello-world:example$KvBulkHelloWorld.java[tag=kv-users,indent=0]
+
+include::java-sdk:hello-world:example$KvBulkHelloWorld.java[tag=kv-bulk-remove,indent=0]
+
+----
+
+{github}
+--
+
+Node.js::
++ 
+--
+Perform a `remove` operation on each document and wait for all the promises to complete before accessing the results.
+
+{kv-bulk-delete-example}
+
+[source,nodejs]
+----
+include::nodejs-sdk:hello-world:example$kv-bulk-hello-world.js[tags=kv-users;kv-bulk-remove,indent=0]
+----
+
+{github}
+--
+
+Python::
++ 
+--
+. Pass some document IDs to remove to the `remove_multi()` function. This will delete the documents from the database.
+
+. A dictionary of `MutationResult` objects is returned.
```

Comment:
Remove number from second paragraph because it's a result, not a "do".
Consequently, remove number from first paragraph, because it then becomes the only step.

```suggestion
Pass the document IDs to remove to the `remove_multi()` function. This will delete the documents from the database.

A dictionary of `MutationResult` objects is returned.
```